packages/superset-ui-plugin-chart-word-cloud/src/WordCloud.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { scaleLinear } from 'd3-scale';
+import cloudLayout, { Word } from 'd3-cloud';
+import { CategoricalColorNamespace } from '@superset-ui/color';
+
+const ROTATION = {
+  flat: () => 0,
+  /* eslint-disable-next-line no-magic-numbers */
+  random: () => Math.floor(Math.random() * 6 - 3) * 30,
+  /* eslint-disable-next-line no-magic-numbers */
+  square: () => Math.floor(Math.random() * 2) * 90,
+};
+
+interface Datum {
+  size: number;
+  text: string;
+}
+
+export interface Props {
+  colorScheme: string;
+  data: Datum[];
+  height: number;
+  rotation: 'flat' | 'random' | 'square';
+  sizeRange: number[];
+  width: number;
+}
+
+interface State {
+  words: Word[];
+}
+
+export default class WordCloud extends React.PureComponent<Props, State> {
+  isMounted: boolean = false;
+  state: State = {
+    words: [],
+  };
+
+  componentDidMount() {
+    this.isMounted = true;
+    this.update();
+  }
+
+  componentDidUpdate() {
+    this.update();
+  }
+
+  componentWillUnmount() {
+    this.isMounted = false;
+  }
+
+  update() {
+    const { data, width, height, rotation, sizeRange } = this.props;
+
+    const size: [number, number] = [width, height];
+    const rotationFn = ROTATION[rotation] || ROTATION.flat;
+
+    const scale = scaleLinear()
+      .range(sizeRange)
+      .domain([0, Math.max(...data.map(d => d.size))]);
+
+    cloudLayout<Datum>()
+      .size(size)
+      .words(data)
+      /* eslint-disable-next-line no-magic-numbers */
+      .padding(5)
+      .rotate(rotationFn)
+      .font('Helvetica')
+      .fontWeight('bold')
+      .fontSize(d => scale(d.size))
+      .on('end', (words: Word[]) => {
+        if (this.isMounted) {
+          this.setState({ words });
+        }
+      })
+      .start();
+  }
+
+  render() {
+    const { width, height, colorScheme } = this.props;
+    const { words } = this.state;
+
+    const colorFn = CategoricalColorNamespace.getScale(colorScheme);
```

Comment:
Do we need the color for the word? since there is no semantic meaning of the color, and to me, it is pretty distracting.

Reply:
I am trying to match current behavior.